src/auth/auth.service.ts

```
@@ -176,7 +180,8 @@ export class AuthService {
         if (strategy) {
             this._refreshSub = strategy.startRefresh(this.config)
                 .filter(grant => !!grant)
-                .subscribe(grant => this.obtainAndLogin(grant));
+                .flatMap(grant => this.obtainAndLogin(grant))
+                .subscribe(result => this.logger.debug('Refresh successful'));
```

I wonder if we should add an error callback too here, and log it. Otherwise, errors are swallowed which might make debugging harder.

---

Yeap, true.